From 77c9aebe6310bf9a4addf14f34615d27d6d1bf94 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 27 Sep 2022 12:46:01 +0300
Subject: [PATCH] [SW-105911] gaudi3: add progress bits

This patch adds additional progress bit for boot info set and get
operations and for slave die before entering wait loop to detect
recovery from master.

Change-Id: I744ac4dc45cdab34a84062d532c20c285709d7c7
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 04a8cc0..019cb7b 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1181,17 +1181,23 @@
 
 	if (!arcboot_rom_global.boot_info_set_done) {
 		if (boot_info_set_no_merge(HL_BOOT_INFO_IMG_DESC_INFO, &p_img_ptr_ctx->desc_info,
-		   sizeof(struct hl_fw_desc_info)))
+		   sizeof(struct hl_fw_desc_info))) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FAIL);
 			return;
+		}
 
 		if (boot_info_set_no_merge(HL_BOOT_INFO_IMG_DESC_DATA,
 		   (uint8_t *)(p_img_ptr_ctx->dest_addr_arc + p_offset[0]),
-		   p_img_ptr_ctx->desc_info.desc_size * p_img_ptr_ctx->desc_info.count))
+		   p_img_ptr_ctx->desc_info.desc_size * p_img_ptr_ctx->desc_info.count)) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_DATA_SET_FAIL);
 			return;
+		}
 
 		if (boot_info_set_no_merge(HL_BOOT_INFO_ROM_SECURITY_DATA, &arcboot_rom_global.sec_data,
-		    sizeof(arcboot_rom_global.sec_data)))
+		    sizeof(arcboot_rom_global.sec_data))) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_SECURITY_DATA_SET_FAIL);
 			return;
+		}
 
 		boot_info_finalize_no_merge();
 
@@ -1201,14 +1207,23 @@
 	} else {
 		if (boot_info_get_no_merge(HL_BOOT_INFO_IMG_DESC_INFO,
 			&p_img_ptr_ctx->desc_info,
-			sizeof(struct hl_fw_desc_info), &copy_sz))
+			sizeof(struct hl_fw_desc_info), &copy_sz)) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FAIL);
 			return;
+		}
+
+		if ((!p_img_ptr_ctx->desc_info.count) || (!p_img_ptr_ctx->desc_info.desc_size)) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_INVALID);
+			return;
+		}
 
 		if (boot_info_get_no_merge(HL_BOOT_INFO_IMG_DESC_DATA,
 			(uint8_t *)(p_img_ptr_ctx->dest_addr_arc),
 			p_img_ptr_ctx->desc_info.desc_size * p_img_ptr_ctx->desc_info.count,
-			&copy_sz))
+			&copy_sz)) {
+			arcboot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_DATA_GET_FAIL);
 			return;
+		}
 
 		for (i = 0; i < p_img_ptr_ctx->desc_info.count; i++) {
 			p_img_desc = (struct hl_fw_img_desc *)(p_img_ptr_ctx->dest_addr_arc +
diff --git a/arch/gaudi3/arc/progress_bits.h b/arch/gaudi3/arc/progress_bits.h
index 12465b1..0351a35 100644
--- a/arch/gaudi3/arc/progress_bits.h
+++ b/arch/gaudi3/arc/progress_bits.h
@@ -35,127 +35,134 @@
 	PROGRESS_VER_LOC = 28,
 	PROGRESS_VER_LOC_0 = 29,
 	PROGRESS_VER_LOC_1 = 30,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 31,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 32,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 33,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 34,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 35,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 36,
-	PROGRESS_BRINGUP_PCI_PRI = 37,
-	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 38,
-	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 39,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 40,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 41,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 42,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 43,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 44,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 45,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 46,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 47,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 48,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 49,
-	PROGRESS_BRINGUP_PCI_SEC = 50,
-	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 51,
-	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 52,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 53,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 54,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 55,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 56,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 57,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 58,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 59,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 60,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 61,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 62,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 63,
-	PROGRESS_ARCBOOT_ROM_TPM = 64,
-	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 65,
-	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 66,
-	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 67,
-	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 68,
-	PROGRESS_BRINGUP_TPM_SPI = 69,
-	PROGRESS_BRINGUP_TPM_SPI_PRI = 70,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_LOADADDR_INVALID = 71,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_STARTADDR_INVALID = 72,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 73,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 74,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 75,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 76,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 77,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 78,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 79,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 80,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 81,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 82,
-	PROGRESS_BRINGUP_TPM_SPI_SEC = 83,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_LOADADDR_INVALID = 84,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_STARTADDR_INVALID = 85,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 86,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 87,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 88,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 89,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 90,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 91,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 92,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 93,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 94,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 95,
-	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 96,
-	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 97,
-	PROGRESS_PPBOOT_START = 98,
-	PROGRESS_PPBOOT_FINISH = 99,
-	PROGRESS_BRINGUP_RAM_PRI = 100,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 101,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 102,
-	PROGRESS_BRINGUP_RAM_SEC = 103,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 104,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 105,
-	PROGRESS_ARCBOOT_RAM_START = 106,
-	PROGRESS_ARCBOOT_RECOVERY = 107,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 108,
-	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 109,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 110,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 111,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 112,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 113,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 114,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 115,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 116,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 117,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 118,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 119,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 120,
-	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 121,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 122,
-	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 123,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 124,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 125,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 126,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 127,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 128,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 129,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 130,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 131,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_OFFSET = 132,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 133,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 134,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 135,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 136,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 137,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 138,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 139,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 140,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 141,
-	PROGRESS_ARCBOOT_EROM_IMG0_LOADADDR_INVALID = 142,
-	PROGRESS_ARCBOOT_EROM_IMG0_STARTADDR_INVALID = 143,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 144,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 145,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 146,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 147,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 148,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 149,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 150,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 151,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 152,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 153,
+	PROGRESS_SLV_DIE_WAITING_ON_RECOVERY_IND = 31,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 32,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 33,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 34,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 35,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 36,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 37,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FAIL = 38,
+	PROGRESS_BOOT_INFO_IMG_DESC_DATA_SET_FAIL = 39,
+	PROGRESS_BOOT_INFO_SECURITY_DATA_SET_FAIL = 40,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FAIL = 41,
+	PROGRESS_BOOT_INFO_IMG_DESC_INFO_INVALID = 42,
+	PROGRESS_BOOT_INFO_IMG_DESC_DATA_GET_FAIL = 43,
+	PROGRESS_BRINGUP_PCI_PRI = 44,
+	PROGRESS_BRINGUP_PCI_PRI_LOADADDR_INVALID = 45,
+	PROGRESS_BRINGUP_PCI_PRI_STARTADDR_INVALID = 46,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 47,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 48,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 49,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 50,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 51,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 52,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 53,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 54,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 55,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 56,
+	PROGRESS_BRINGUP_PCI_SEC = 57,
+	PROGRESS_BRINGUP_PCI_SEC_LOADADDR_INVALID = 58,
+	PROGRESS_BRINGUP_PCI_SEC_STARTADDR_INVALID = 59,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 60,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 61,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 62,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 63,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 64,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 65,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 66,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 67,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 68,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 69,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 70,
+	PROGRESS_ARCBOOT_ROM_TPM = 71,
+	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 72,
+	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 73,
+	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 74,
+	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 75,
+	PROGRESS_BRINGUP_TPM_SPI = 76,
+	PROGRESS_BRINGUP_TPM_SPI_PRI = 77,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_LOADADDR_INVALID = 78,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_STARTADDR_INVALID = 79,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 80,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 81,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 82,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 83,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 84,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 85,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 86,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 87,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 88,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 89,
+	PROGRESS_BRINGUP_TPM_SPI_SEC = 90,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_LOADADDR_INVALID = 91,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_STARTADDR_INVALID = 92,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 93,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 94,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 95,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 96,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 97,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 98,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 99,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 100,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 101,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 102,
+	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 103,
+	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 104,
+	PROGRESS_PPBOOT_START = 105,
+	PROGRESS_PPBOOT_FINISH = 106,
+	PROGRESS_BRINGUP_RAM_PRI = 107,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 108,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 109,
+	PROGRESS_BRINGUP_RAM_SEC = 110,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 111,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 112,
+	PROGRESS_ARCBOOT_RAM_START = 113,
+	PROGRESS_ARCBOOT_RECOVERY = 114,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 115,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 116,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 117,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 118,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 119,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 120,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 121,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 122,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 123,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 124,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 125,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 126,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 127,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 128,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 129,
+	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 130,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 131,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 132,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 133,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 134,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 135,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 136,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 137,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 138,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_OFFSET = 139,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 140,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 141,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 142,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 143,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 144,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 145,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 146,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 147,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 148,
+	PROGRESS_ARCBOOT_EROM_IMG0_LOADADDR_INVALID = 149,
+	PROGRESS_ARCBOOT_EROM_IMG0_STARTADDR_INVALID = 150,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 151,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 152,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 153,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 154,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 155,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 156,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 157,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 158,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 159,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 160,
 };
diff --git a/arch/gaudi3/arc/slave.c b/arch/gaudi3/arc/slave.c
index cdfa446..a237397 100644
--- a/arch/gaudi3/arc/slave.c
+++ b/arch/gaudi3/arc/slave.c
@@ -31,6 +31,8 @@
 	if (!arcboot_rom_global.die_id)
 		return;
 
+	arcboot_rom_status_update(PROGRESS_SLV_DIE_WAITING_ON_RECOVERY_IND);
+
 	while (1) {
 		val = regmap_readl(mmD0_PSOC_GLOBAL_CONF_DIE_RECOVERY_IN);
 		if (val & D0_PSOC_GLOBAL_CONF_DIE_RECOVERY_IN_IND_M)
